packages/analytics-core/src/network-observer.ts

```diff
@@ -43,15 +80,25 @@ export class NetworkObserver {
     }
     this.eventCallbacks.set(eventCallback.id, eventCallback);
     if (!this.isObserving) {
+      /* istanbul ignore next */
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      const originalXhrOpen = this.globalScope?.XMLHttpRequest?.prototype?.open;
+      /* istanbul ignore next */
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      const originalXhrSend = this.globalScope?.XMLHttpRequest?.prototype?.send;
```

Comment:
Looks like we never restore to origin. I understand that override won't persist when page reload (jump to another website). But just thinking about some edge cases here.